### PR TITLE
refactor: service to be stateless

### PR DESCRIPTION
**Pull request by cmyui**
Co-authored-by: RealistikDash <realistikdash@gmail.com>

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -1,9 +1,9 @@
 #!/usr/bin/env python3.9
 from __future__ import annotations
 
-import ddtrace
+# import ddtrace
 
-ddtrace.patch_all()
+# ddtrace.patch_all()
```
Comment (cmyui) [inline main.py]: @cmyui this for local only

File: main.py [hunk 2]
```diff
@@ -150,19 +150,14 @@ def make_app():
             )
             raise
 
-        # Get build date for login notifications
-        with open("build.date") as f:
-            timestamp = dt.utcfromtimestamp(int(f.read()))
-            glob.latestBuild = timestamp.strftime("%b %d %Y")
-
```
Comment (cmyui) [inline main.py:156]: deleted these bad boys for stateless; need to remember to remove from sql @cmyui

File: objects/stream.py [hunk 3]
```diff
@@ -9,99 +9,115 @@
...
+) -> None:
+    """
+    Add a client to this stream if not already in
+
+    :param client: client (osuToken) object
+    :param token: client uuid string
+    :return:
+    """
+    if not (client or token):
+        return
```
Comment (cmyui) [inline objects/stream.py]: want to get rid of these ideally; check needs to be done before getting here

File: objects/streamList.py [hunk 4]
```diff
@@ -10,112 +10,132 @@
 
     from objects.osuToken import token
 
+
+# def __init__(self):
+#     self.streams = {}
+
```
Comment (cmyui) [inline objects/streamList.py]:
```suggestion
```

File: objects/streamList.py [hunk 5]
```diff
@@ -10,112 +10,132 @@
...
+        current_clients = stream.getClients(name)
+        for i in current_clients:
+            if i in glob.tokens.tokens:
+                glob.tokens.tokens[i].leaveStream(name)
+
+        #self.streams.pop(name)
+        previous_members = stream.getClients(name)
+        for token in previous_members:
+            stream.removeClient(name, token=token)
+        glob.redis.srem(make_key(), name)
```
Comment (cmyui) [inline objects/streamList.py]: not 100%

File: objects/channelList.py [hunk 6]
```diff
@@ -3,162 +3,239 @@
...
+            continue
+
+        glob.redis.sadd("bancho:channels", chan["name"])
+        glob.redis.set(make_key(chan['name']), json.dumps({
+            "name": chan["name"],
+            "description": chan["description"],
+            "public_read": chan["public_read"] == 1,
+            "public_write": chan["public_write"] == 1,
+            "moderated": False,
+            "instance": False ,# all db tables are not
```
Comment (cmyui) [inline objects/channelList.py]:
```suggestion
            "instance": False, # channels stored persistently are never instance channels
```

File: objects/fokabot.py [hunk 7]
```diff
@@ -27,18 +27,19 @@
...
+        streamList.broadcast("main", serverPackets.userStats(999))
 
 
 def disconnect() -> None:
     with glob.tokens:
         token = glob.tokens.getTokenFromUserID(999)
         assert token is not None
 
-        glob.tokens.deleteToken(token)
+        glob.tokens.deleteToken(token.token)
```
Comment (cmyui) [inline objects/fokabot.py]: bugfix (probably old 6y/o bug from ripple?)

File: objects/streamList.py [hunk 8]
```diff
@@ -10,112 +10,127 @@
...
+
+    :param name: name of the stream
+    :param args:
+    :param kwargs:
+    :return:
+    """
+
+    streams = getStreams()
+    if name in streams:
+        stream.dispose(name, *args, **kwargs)
```
Comment (cmyui) [inline objects/streamList.py:130]: probably can replace these with the stream equivalents? or remove them from stream? (maybe that's more consistent with channelList)

**Review by tsunyoku — APPROVED** [review]
wow.

**Pull request merged**